### PR TITLE
Fix architecture preference for MacOS game shortcuts

### LINKED ISSUE #6039: [Bug] Application shortcut reduces performance (MacOS)
### Description of the issue

There is a functionality to create an app shortcuts. Games started by using this shortcut perform worse comparing to start from Ryujinx UI itself.

Screenshot 2023-12-16 at 03 14 05

### Reproduction steps

1. Create application shortcut
2. Compare performance between two scenarios:
 - Opening Ryujinx and running game from the UI
 - Running game from Application Shortcut
 
 <details>
  <summary>Game started from UI: 60+ FPS</summary>
  
Screenshot 2023-12-16 at 03 19 09
  
</details>
 
  <details>
  <summary>Game started from application shortcut: < 30 FPS</summary>
  
Screenshot 2023-12-16 at 03 20 53
  
</details>

Even shaders compilation speed before game start is different. 

UPD.: I have checked Activity Monitor and difference is: **application shortcut starts Intel app process instead of native Apple Silicon** 

### Log file

[Ryujinx_1.1.1100_2023-12-16_03-19-35.log](https://github.com/Ryujinx/Ryujinx/files/13690341/Ryujinx_1.1.1100_2023-12-16_03-19-35.log)

### OS

MacOS Sonoma 14.2

### Ryujinx version

1.1.1100

### CPU

MacBook M3 Pro

### RAM

18 GB

### PR DESCRIPTION
**Pull request by Alex0007**
Should fix #6039 

tldr: previously MacOS game shortcut was starting Ryujinx via Rosetta translator (huge performance hit)

### CODE REVIEW

**Comment by TSRBerry** [comment]
Could that also be adjusted using the plist file? It's odd to me that this env var seems to be required.

**Comment by Alex0007** [comment]
@TSRBerry 

I've tried `LSArchitecturePriority` and `LSRequiresNativeExecution` (from [this](https://stackoverflow.com/a/68208374/1518708) answer), but they didn't worked 😐

File: src/Ryujinx.Ui.Common/Helper/ShortcutHelper.cs [hunk 1]
```diff
@@ -69,7 +69,7 @@ private static void CreateShortcutMacos(string appFilePath, byte[] iconData, str
             using StreamWriter scriptFile = new(scriptPath);
 
             scriptFile.WriteLine("#!/bin/sh");
```
Comment (jcm93) [inline src/Ryujinx.Ui.Common/Helper/ShortcutHelper.cs:71]:
```suggestion
            string launchScript = $"""
                                   #!/bin/sh
                                   launch_arch="$(uname -m)"
                                   if [ "$(sysctl -in sysctl.proc_translated)" = "1" ]
                                   then
                                     launch_arch="arm64"
                                   fi

                                   arch -$launch_arch {basePath} {GetArgsString(appFilePath)}
                                   """;
            scriptFile.Write(launchScript);
```

What an annoying macOS bug. This is ugly but avoids setting an environment variable, as an alternative.

File: src/Ryujinx.Ui.Common/Helper/ShortcutHelper.cs [hunk 2]
```diff
@@ -68,8 +68,17 @@ private static void CreateShortcutMacos(string appFilePath, byte[] iconData, str
...
+                                   #!/bin/sh
+                                   launch_arch="$(uname -m)"
+                                   if [ "$(sysctl -in sysctl.proc_translated)" = "1" ]
+                                   then
+                                     launch_arch="arm64"
+                                   fi
+
+                                   arch -$launch_arch {basePath} {GetArgsString(appFilePath)}
+                                   """;
+            scriptFile.Write(launchScript);
```
Comment (TSRBerry) [inline src/Ryujinx.Ui.Common/Helper/ShortcutHelper.cs:81]: Now that this script is a little larger, could you turn this into an `EmbeddedResource` like `shortcut-template.plist` above for example?

**Review by TSRBerry — COMMENTED** [review]
Lgtm!

**Review by marysaka — APPROVED** [review]
lgtm, but same comment as @TSRBerry here

**Review by TSRBerry — APPROVED** [review]

**Comment by NitroTears** [comment]
Good work, thank you!
